discovery/kubernetes/endpoints.go

```diff
@@ -305,7 +305,12 @@ func (e *Endpoints) buildEndpoints(eps *apiv1.Endpoints) *targetgroup.Group {
 		}
 
 		if e.withNodeMetadata {
-			target = addNodeLabels(target, e.nodeInf, e.logger, addr.NodeName)
+			if addr.NodeName != nil {
+				target = addNodeLabels(target, e.nodeInf, e.logger, addr.NodeName)
+			}
+			if addr.TargetRef != nil && addr.TargetRef.Kind == "Node" {
```

Comment:
Is there a case where both of these conditions would be true?

Reply:
As far as I know, the expected behaviour is that it can't but I could be explicit here just to be safe